### ISSUE TITLE
Manual file resolution in vite `nx-tsconfig-paths.plugin` does not take index/directory imports into account

### ISSUE THREAD

**Issue by JanPretzel** [nrwl/nx#19028]
### Current Behavior

When exposing a TypeScript path with a placeholder in `tsconfig.base.json` the `nx-tsconfig-paths.plugin` manual file resolution implemented in #18477 does not take index/directory imports into account.

Suppose the path looks like this `"@my/components/*": ["libs/components/src/lib/*"]`  and the lib folder structure is something like this

```
lib/
├── button/
│   ├── index.tsx
├── radio/
│   ├── radio.tsx
```

With the fix from #18477 one can easily directly import the radio component but not the button component.

```ts
// this import can be resolved
import Radio from '@my/components/radio/radio';

// this import cannot be resolved
import Button from '@my/components/button';
```

### Expected Behavior

The file resolution should be able to identify index/folder imports and resolve those.

### Steps to Reproduce

1. Create a React application with vite as bundler
2. Create a React library without bundler
3. In the library create a Folder, e.g. called `button` and put a component in a file called `index.tsx`
4. In `tsconfig.base.json` manually add a path with placeholder to allow importing directly from within the libraries `lib` folder, e.g. `"@my/<your-library-name>/*": ["libs/<your-library-name>/src/lib/*"]`
5. In the React application that you previously created try to import the component previously created, e.g. `import Button from '@my/<your-library-name>/button'`

### Nx Report

```shell
Node   : 16.20.2
   OS     : linux-x64
   npm    : 9.8.1
   
   nx                 : 16.8.0
   @nx/js             : 16.8.0
   @nx/jest           : 16.8.0
   @nx/linter         : 16.8.0
   @nx/workspace      : 16.8.0
   @nx/devkit         : 16.8.0
   @nx/eslint-plugin  : 16.8.0
   @nx/plugin         : 16.8.0
   @nx/react          : 16.8.0
   @nrwl/tao          : 16.8.0
   @nx/vite           : 16.8.0
   @nx/web            : 16.8.0
   nx-cloud           : 16.3.0
   typescript         : 5.2.2
   ---------------------------------------
```

### Package Manager Version

npm 9.8.1

### Operating System

- [ ] macOS
- [X] Linux
- [ ] Windows
- [ ] Other (Please specify)

**Comment by svallory** [comment]
Just found out that if you override the `paths` property in the project's `tsconfig` file they are simply ignored

**Comment by JanPretzel** [comment]
> Just found out that if you override the `paths` property in the project's `tsconfig` file they are simply ignored

I don't think that this has anything to do with what I reported, since I am manually editing the `paths` property in the `tsconfig.base.json`. In general this works, but only index/folder imports cannot be resolved.

**Comment by svallory** [comment]
> I am manually editing the `paths` property in the `tsconfig.base.json`

You are right! I must have read in rush and missed the fact that you were editing the `tsconfig.base`  and not the one in your library. This is definitely a bug.

If you are looking for a solution in the mean term though, I fixed my issue by adding the paths I wanted to `libs/<my lib>/tsconfig.json` and using [vite-tsconfig-paths](https://www.npmjs.com/package/vite-tsconfig-paths) alongside `nx-tsconfig-paths`

**Comment by JanPretzel** [comment]
@svallory thanks for the hint! Will try that if the fix will take longer.

**Comment by EugenMayer** [comment]
Is there any chance to move this issue forward? We do rely on the fix #19030 and it actually hinders us in using the generators.

Is there any way we can provide more information or is there anything missing here? Any feedback to the PR would be really appropriated. Thanks!

**Comment by lukascivil** [comment]
Hi guys, any updates here?
